File: String 1/12. SCHAR/SCHAR.py
import sys
sys.stdin = open('BAI3.inp', 'r')
sys.stdout = open('BAI3.out', 'w')
t = int(input())

def cpp (a: str):
	h = [0]*26
	for i in a:
		h[ord(i) - ord('a')] += 1
	return h

r = ''
for i in range (1, t+1):
	s = input(); hmm = cpp(s); f = 1;
	for n in range(len(s)):
		if hmm[ord(s[n]) - ord('a')] == 1:
			r += str(n+1) + '\n'
			f = 0
			break
	if f:
			r += '-1\n'
print(r)

File: String 1/12. SCHAR/SCHAR.cpp
#include <bits/stdc++.h>
using namespace std;

long long t, h[26], f;
string s;

int main () {
	ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
	freopen("BAI3.inp", "r", stdin);
	freopen("BAI3.out", "w", stdout);
	cin >> t;

	for (int i = 0; i < t; i++) {
		cin >> s;
		fill(h, h+26, 0);
		f = 1;
		for (int j = 0; j < s.length(); j++) h[s[j]-'a']++;
		for (int j = 0; j < s.length(); j++) {
			if (h[s[j]-'a'] == 1) {
				cout << j+1 << '\n';
				f = 0;
				break;
			}
		}
		if (f) cout << -1 << '\n';
	}
}